javaslang/src/main/java/javaslang/Lazy.java

```diff
@@ -94,7 +94,7 @@ private Lazy(Supplier<? extends T> supplier) {
      */
     public static <T> Lazy<Seq<T>> sequence(Iterable<? extends Lazy<? extends T>> values) {
         Objects.requireNonNull(values, "values is null");
-        return Lazy.of(() -> List.ofAll(values).map(Lazy::get));
+        return Lazy.of(() -> List.ofAll(values).map(l -> l.get()));
```

Comment:
It think it may be the case because of the generic bounds of `values`...

Type-investigation seems to got worse in Java 9. These are the WTF!? moments during development - we know that the type is right but the compiler does complain.

Starting from JDK 1.8.0_40 the compiler got steadily better. It is sad to see that Java 9 is a step back :(

---

However, I'm not sure if I *want* to use a lambda here. Instead JDK 9 should be fixed :) That's also the case for other changes. Is it possible to defer some changes an let the build break? Maybe we should report these finding and hope that they get fixed until the release of JDK 9 (which may be [delayed again](http://mail.openjdk.java.net/pipermail/jdk9-dev/2016-September/004820.html)??).

If we let the JDK9 build break it should not affect the JDK8 build of our PRs, merges and so on...